### DISCUSSION TITLE
Marking multi-booking as paid with offline payment

### DISCUSSION THREAD

**Discussion by markrdavidson** [liedekef/events-made-easy#20] [Q&A] [answered]
Hey Franky,

My client accepts offline payments for their events.  When I send my client a registration notice by email, I would like to be able to include a "Mark As Paid" link so they don't have to go searching through the "Approved Registrations" list to select each booking id.

Is there a way to mark ALL the seats in a multiple booking as "paid" all at once?

Is there a way I can link to  /wp-admin/admin.php?page=eme-registration-seats  with the seat(s) checkboxes preselected?

I had a look at providing links to each booking id to edit, but the link is always "expired", as you will understand.

Your guidance is appreciated!
Mark

**Comment by liedekef** [comment, marked as answer]
In the "Approved registrations" you can search on payment id, that allows them to find all bookings linked to a payment, select them all in 1 go and "mark as paid" then. The payment id you can pass along via #_PAYMENTID

**Reply by markrdavidson** [reply]
Possibly this is something you could consider for a future release - an easier "Mark As Paid" link that could be included in an email to the event administrator... for single OR multi-seat events.  Thanks for considering this!

Mark

**Reply by liedekef** [reply]
The only thing I could do is allow a paymentid-param to be taken as argument to the booking approval/registration pages. While that is possible, it will only be 1 step less to do. The bookings would still need to be selected and "marked as paid", although the whole process would be slightly simpler for a basic admin.
Of course this is something you can achieve using some custom jquery too: check if the param is set, fill out the value of the corresponding filter field and call the filter button.

**Reply by markrdavidson** [reply]
It would actually save THREE steps, based on the instructions I provide to my client in the email notification... your suggestion would get us straight to step FOUR.  =)

If this registration is paid offline, follow these steps to record the payment:
1) [Click here] (https://www.....com/wp-admin/admin.php?page=eme-registration-seats) for booking administration.
2) Click "Show/hide extra filters".
3) In the "Filter on payment id" box enter Payment Id " #_PAYMENTID " and click "Filter bookings".
4) Click the checkbox in the header row to select all matching bookings.
5) Choose "Bulk Actions" > "Mark as paid", and click "Apply".

**Reply by markrdavidson** [reply]
Maybe you could also have "selectall=true" as a parameter / argument too?  ;)

**Reply by liedekef** [reply]
Feel free to create a feature request for this in the "Issues"  next time.
See this change: https://github.com/liedekef/events-made-easy/commit/a612110674187ed1d0237dc88a2d768723dc31f3
selectall=true is something totally else though (it acts on the javascript already used for the main select, so not done here)

**Reply by liedekef** [reply]
Btw: the next version will also have a "Mark paid" button in the table for approved bookings too, similar to the "Mark paid and approve" button in the table for pending bookings. So one step less to do then :-)
